docs/diagrams/models/ModelClassDiagram.puml

```diff
@@ -0,0 +1,81 @@
+@startuml
+!include ../style.puml
+skinparam arrowThickness 1.1
+skinparam arrowColor MODEL_COLOR
+skinparam classBackgroundColor MODEL_COLOR
+skinparam linetype ortho
+
+note as n1
+We omit specific details of models
+Resident, Room, ResidentRoom, Issue,
+CommandHistorymand Alias as they
+areexplained in greater detail in their
+specific sections.
+
+Due to PlantUML restrictions, the
+interaction that ModelManager and
+StatefulAddressBook have with the
+packages is shown in separate
+zoomed-in diagrams.
+end note
+
+Package Model <<Rectangle>>{
+    together overarching {
+        Interface ReadOnlyAddressBook <<Interface>>
+        Interface Model <<Interface>>
+        Interface ObservableList <<Interface>>
+        Class AddressBook
+        Class StatefulAddressBook
+        Class ReadOnlyAddressBook
+        Class Model
+        Class ModelManager
+        Class UserPrefs
+        Class ReadOnlyUserPrefs
+    }
+
+    together res {
+        Package Resident {
+        }
+        Package Room {
+        }
+        Package ResidentRoom {
+        }
+        Package Issue {
+        }
+        Package Alias {
+        }
+        Package CommandHistory {
+        }
+    }
+}
+' ===== Overarching stuff =====
+Class HiddenOutside #FFFFFF
+HiddenOutside ...> Model
+
+StatefulAddressBook -left-|> AddressBook
+AddressBook .left.|> ReadOnlyAddressBook
+StatefulAddressBook -up-> "1..*" ReadOnlyAddressBook
+
+ModelManager .up.|> Model
+Model .left.> ObservableList
+ModelManager *-left-> "1" StatefulAddressBook
+ModelManager *-right-> "1" UserPrefs
+UserPrefs .up.|> ReadOnlyUserPrefs
+
+
+ModelManager *-down-> ResidentRoom
+ModelManager *-down-> Resident
+ModelManager *-down-> Room
+
+ModelManager *-> Issue
+ModelManager *-> CommandHistory
+ModelManager *-> Alias
+
+StatefulAddressBook ---down-> Resident
+StatefulAddressBook ---down-> Room
+StatefulAddressBook ---down-> ResidentRoom
+StatefulAddressBook ---down-> Issue
+StatefulAddressBook ---down-> Alias
```

Comment:
Hmm, aren't the associations to the various sub-components from `AddressBook`  rather than `StatefulAddressBook`?

Also, I think the `AddressBook` has exactly one of each.